components/feature/addons/src/main/java/mozilla/components/feature/addons/amo/AddonCollectionProvider.kt

```diff
@@ -176,6 +185,22 @@ class AddonCollectionProvider(
         }
     }
 
+    /**
+     * Deletes cache files from previous (now unused) collections.
+     */
+    @VisibleForTesting
+    internal fun deleteUnusedCacheFiles() {
+        val currentCacheFileName = getCacheFileName()
```

Comment:
We don't need to use our lock here as we're not touching files that are currently in use.